File: cdap-ui/app/services/namespace.js
/*
 * Copyright © 2015 Cask Data, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */

angular.module(PKG.name + '.services')
  .service('myNamespace', function myNamespace($q, MyCDAPDataSource, EventPipe, $http, $rootScope, myAuth, myHelpers, $state) {

    this.namespaceList = [];
    window.CaskCommon.StatusFactory.startPollingForBackendStatus();
    var data = new MyCDAPDataSource(),
        prom,
        queryInProgress = null;


    this.getList = function (force) {
      if (!force && this.namespaceList.length) {
          return $q.when(this.namespaceList);
      }

      if (!queryInProgress) {
        prom = $q.defer();
        queryInProgress = true;
        data.request(
          {
            _cdapPath: '/namespaces',
            method: 'GET'
          })
            .then(
              (function(res) {

                if (!res.length && !$state.includes('admin.**')) {
                  $state.go('unauthorized');
                }

                this.namespaceList = res;
                EventPipe.emit('namespace.update');
                prom.resolve(res);
                queryInProgress = null;
              }).bind(this),
              function (err) {
                prom.reject(err);
                queryInProgress = null;
              }
            );
      }

      return prom.promise;
    };

    this.getDisplayName = function(name) {
      var ns = this.namespaceList.filter(function(namespace) {
        return namespace.name === name;
      });
      return ns[0].name || name;
    };

  });

File: cdap-app-fabric/src/main/java/io/cdap/cdap/graphql/objects/Namespace.java
/*
 *
 * Copyright © 2019 Cask Data, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */

package io.cdap.cdap.graphql.objects;

/**
 *
 */
public class Namespace {

  private final String name;
  private final String description;
  private final long generation;

  private Namespace(Builder builder) {
    this.name = builder.name;
    this.description = builder.description;
    this.generation = builder.generation;
  }

  public String getName() {
    return name;
  }

  public String getDescription() {
    return description;
  }

  public long getGeneration() {
    return generation;
  }

  /**
   * TODO
   */
  public static class Builder {

    private String name;
    private String description;
    private long generation;

    public Builder name(String name) {
      this.name = name;

      return this;
    }

    public Builder version(String description) {
      this.description = description;

      return this;
    }

    public Builder scope(long generation) {
      this.generation = generation;

      return this;
    }

    public Namespace build() {
      return new Namespace(this);
    }
  }
}
